packages/components/src/InputTime/InputTime.rebuilt.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ChangeEvent, useId, useRef } from "react";
+import { useTimePredict } from "./hooks/useTimePredict";
+import { InputTimeProps, InputTimeRebuiltProps } from "./InputTime.Types";
+import { dateToTimeString, timeStringToDate } from "./utils/input-time-utils";
+import { FormFieldWrapper, useFormFieldWrapperStyles } from "../FormField";
+
+export function InputTimeRebuilt({
+  defaultValue,
+  value,
+  onChange,
+  ...params
+}: InputTimeRebuiltProps) {
+  const ref =
+    (params.inputRef as React.RefObject<HTMLInputElement>) ??
+    useRef<HTMLInputElement>(null);
```

Comment:
either use the ref we were provided via `inputRef` or use our own instance to avoid ever having 2 separate instances where one would never be set